src/main/kotlin/biokotlin/genome/MAFProcessingUtils.kt

```diff
@@ -447,4 +452,321 @@ fun getCoverageIdentityPercentForMAF(mafFile:String, region:String = "all"): Dat
     }.toDataFrame()
     return df
 }
+/**
+ * function to bgzip and create tabix index on a gvcf file
+ *
+ * This method uses the -f option on  bgzip to overwrite any existing files
+ * This handles the case where there already exists a bgzipped file, however
+ * it is still required to have the non-bzipped file present.
+ *
+ * @return the filename of the bgzipped gvcf
+ */
+fun compressAndIndexFile(fileName: String): String {
+
+    try {
+        // First bgzip the file
+
+        // bgzip the file - needed to create index
+        // use the -f option to overwrite any existing file
+        println("bgzipping  file ${fileName}")
+        val gvcfGzippedFile = fileName + ".gz"
+        var builder = ProcessBuilder("conda","run","-n","phgv2-conda",
+            "bgzip", "-f", fileName)
+
+        var process = builder.start()
+        var error: Int = process.waitFor()
+        if (error != 0) {
+            println("\nERROR $error creating bgzipped  version of file: $fileName")
+            throw IllegalStateException("bgzipAndIndexGVCFfile: error trying to bgzip file ${fileName}: ${error}")
+        }
+
+        // File has been gzipped, now index it.
+        // Use the -f option to overwrite any existing index
+        // We will use bcftools to create the csi index
+        // ORiginal PHG used tabix, we wnat csi indexes to allow for large genomes e.g wheat.
+        // TileDB supports .csi indexed files.
+        builder = ProcessBuilder("conda","run","-n","phgv2-conda","bcftools", "index", "-c",gvcfGzippedFile)
+        process = builder.start()
+        error = process.waitFor()
+        if (error != 0) {
+            println("\nERROR $error creating tabix indexed  version of file: $gvcfGzippedFile")
+            throw IllegalStateException("bgzipAndIndexGVCFfile: error trying to run bcftools index -c on file ${gvcfGzippedFile}: ${error}")
```

Comment:
Should this say compressAndIndexFile? And the one below

Reply:
Yes - changed them all.